shell/platform/android/test/io/flutter/plugin/editing/TextInputPluginTest.java

```diff
@@ -155,6 +160,34 @@ public void setTextInputEditingState_nullInputMethodSubtype() {
         assertEquals(1, testImm.getRestartCount(testView));
     }
 
+    @Test
+    public void inputConnection_createsActionFromEnter() {
+        Log.setLogLevel(android.util.Log.VERBOSE);
+        TestImm testImm = Shadow.extract(RuntimeEnvironment.application.getSystemService(Context.INPUT_METHOD_SERVICE));
+        FlutterJNI mockFlutterJni = mock(FlutterJNI.class);
+        View testView = new View(RuntimeEnvironment.application);
+        DartExecutor dartExecutor = spy(new DartExecutor(mockFlutterJni, mock(AssetManager.class)));
+        TextInputPlugin textInputPlugin = new TextInputPlugin(testView, dartExecutor, mock(PlatformViewsController.class));
+        textInputPlugin.setTextInputClient(0, new TextInputChannel.Configuration(false, false, true, TextInputChannel.TextCapitalization.NONE, new TextInputChannel.InputType(TextInputChannel.TextInputType.TEXT, false, false), null, null));
+        // There's a pending restart since we initialized the text input client. Flush that now.
+        textInputPlugin.setTextInputEditingState(testView, new TextInputChannel.TextEditState("", 0, 0));
+
+
+        ByteBuffer message = JSONMethodCodec.INSTANCE.encodeMethodCall(
+            new MethodCall("TextInputClient.performAction", Arrays.asList(0, "TextInputAction.done")));
```

Comment:
The strings in this method call can be anything: verify doesn't care.  Any idea how I can get it to care?

Reply:
I would have expected that `verify` was using the `equals` method on the ByteBuffer, that's not the case?  Mockito has ArgumentMatchers so you can tweak the logic it uses (https://www.baeldung.com/mockito-argument-matchers).  I wouldn't have expected that you'd need it.

Reply:
But do argument matchers work with spies, or only with mocks? I thought those were for mocks.

Reply:
Ahh, I found a way: https://stackoverflow.com/questions/10287746/mockito-spy-test

I'll do that.

Reply:
Alright, it wasn't really all that simple, but check it out and let me know what you think.